### PR TITLE
fix: add boolean number to datastorepredicateobject operand

### LINKED ISSUE #3635: DataStorePredicateObject.operand type needs to support other types other than string
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

CRA

### What browsers are you seeing the problem on?

Chrome

### Please describe your bug.

Trying to create a valid predicate object like below isn't possible in TS because the type only takes strings for operand. 
```
{
  field:  "isActive",
  operand: false,
  operator: "eq",
  or: undefined,
}
```

### What's the expected behaviour?

Numbers and booleans should be valid.
```
{
  field:  "isActive",
  operand: false,
  operator: "eq",
  or: undefined,
}
```

### Help us reproduce the bug!

Try to use a boolean value for operand and pass it into the `createDataStorePredicate` function and you will see TS errors.

### Code Snippet

```typescript
// Put your code below this line.
createDataStorePredicate(
  {
    field:  "isActive",
    operand: false,
    operator: "eq",
    or: undefined,
  }
)
```

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by letsbelopez**
#### Description of changes

Add boolean and number to the operand type.

#### Issue #, if available

fixes #3635 

#### Description of how you validated changes

I added a unit test.

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/types/datastore.ts [hunk 1]
```diff
@@ -44,6 +44,6 @@ export type DataStorePredicateObject = {
   and?: DataStorePredicateObject[];
   or?: DataStorePredicateObject[];
   field?: string;
-  operand?: string;
+  operand?: string | boolean | number;
```
Comment (calebpollman) [inline packages/react/src/primitives/types/datastore.ts:47]: Is there a type from `DataStore` that we could use here?

Comment (letsbelopez) [reply packages/react/src/primitives/types/datastore.ts:47]: as far as I can tell, datastore is using `any` for operand type. So I'd say no.

Comment (calebpollman) [reply packages/react/src/primitives/types/datastore.ts:47]: I mean, `any` is a type right? 😅

**Review by calebpollman — APPROVED** [review]

File: packages/react/src/primitives/types/datastore.ts (hunk 1 above)
Comment (reesscot) [reply packages/react/src/primitives/types/datastore.ts:47]: Yeah, looks like this is the PredicateObject type:

```
export type PredicateObject<T extends PersistentModel> = {
	field: keyof T;
	operator: keyof AllOperators;
	operand: any;
};
```
https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/types.ts#L780

We've tested that `boolean` and `number` work correctly I presume?

**Review by reesscot — DISMISSED** [review]
LGTM

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by reesscot — CHANGES_REQUESTED** [review]
Blocking this for now due to other changes going out next week

**Review by reesscot — APPROVED** [review]
LGTM

**Pull request merged**